engine/events.go

```diff
@@ -308,10 +308,10 @@ func EventManger() {
 
 // IsValidExchange validates the exchange
 func IsValidExchange(exchangeName string) bool {
-	cfg := config.GetConfig()
-	for x := range cfg.Exchanges {
-		if strings.EqualFold(cfg.Exchanges[x].Name, exchangeName) && cfg.Exchanges[x].Enabled {
-			return true
+	exchanges := config.GetConfig().GetAllExchangeConfigs()
```

Comment:
a possibly better approach (to take care of the race problem) is to eliminate all the uses of `config.GetConfig()` and always get it from the *Engine object instead